agg-ast/schema_derivation/src/schema_derivation_tests/stage.rs

```diff
@@ -7,6 +7,62 @@ use mongosql::{
 };
 use std::collections::BTreeMap;
 
+mod add_fields {
+    use super::*;
+
+    test_derive_stage_schema!(
+        add_fields,
```

Comment:
I took a stab at this as part of SQL-2366, had a couple test cases in mind that might be worth adding: https://github.com/mongodb/mongosql/pull/37/files#diff-99e52b8280797b2fa94ad9864f86606367010126938e62b9d30fe2e3525d62d9

I think the add fields with $$REMOVE and add fields with a nested field would be the ones I'm most interested in adding here (I just took the examples from the docs and tested each of them)